ui/src/components/ConfigurationForm/ConfigurationForm.component.tsx

```diff
@@ -57,20 +58,18 @@ export const ConfigurationForm: React.FC<ConfigurationFormProps> = ({ isEmbedded
   const isEffectImplementationSelectable = Boolean(effectType) && Boolean(scalaVersion);
 
   useEffect(() => {
-    // state is passed from preview starter, after the click on 'Back' button.
-    if (location.state !== null) {
-      const request = location.state as StarterRequest;
-      form.setValue('projectName', request.projectName);
-      form.setValue('groupId', request.groupId);
-      form.setValue('effect', request.effect);
-      form.setValue('implementation', request.implementation);
-      form.setValue('addDocumentation', request.addDocumentation);
-      form.setValue('addMetrics', request.addMetrics);
-      form.setValue('json', request.json);
-      form.setValue('scalaVersion', request.scalaVersion);
-      form.setValue('builder', request.builder);
+    if (formData !== undefined) {
+      form.setValue('projectName', formData.projectName);
```

Comment:
Could we simplify this a bit?
```
Object.entries(formData).map(([key, value]) => form.setValue(key, value));
```

Reply:
It looks nice, but typescript complains. It has inferred type `'scalaVersion'|'builder'|...` and won't allow plain `string` :(.